# -*- coding: utf-8 -*-
# Generated by Django 1.10.3 on 2017-05-04 11:59
from __future__ import unicode_literals

from django.db import migrations
from django.db.migrations.exceptions import InvalidMigrationPlan


class Migration(migrations.Migration):

    dependencies = [
        ('scipost', '0050_auto_20170416_2152'),
    ]

    operations = [
        migrations.AlterModelTable(
            name='Feedback',
            table='scipost_feedback',
        ),
        migrations.AlterModelTable(
            name='Motion',
            table='scipost_motion',
        ),
        migrations.AlterModelTable(
            name='Nomination',
            table='scipost_nomination',
        ),
        migrations.AlterModelTable(
            name='VGM',
            table='scipost_vgm',
        ),
        migrations.RemoveField(
            model_name='remark',
            name='feedback',
        ),
        migrations.RemoveField(
            model_name='remark',
            name='motion',
        ),
        migrations.RemoveField(
            model_name='remark',
            name='nomination',
        ),
    ]

    def apply(self, *args, **kwargs):
        '''Forward apply the migration only if user explictly accepts the migration.'''
        print('\n\nDue to earlier migration issues, this migration will remove the Remark foreignkey'
              ' data to `Feedback`, `Motion` and `Nomination`. Please make a backup to'
              ' be able to *manually* fix the relations afterwards.')
        go_ahead = input(' Do you want to continue? [y/N] ')
        if go_ahead == 'y':
            return super().apply(*args, **kwargs)
        else:
            raise InvalidMigrationPlan
